backend/Server/Server.java

```diff
@@ -0,0 +1,279 @@
+import java.io.File;
+import java.io.OutputStream;
+import java.io.FileInputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.URLDecoder;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+// Used for testing to create shallow JSON strings. -Kyle
+import java.lang.StringBuilder;
+
+import static Server.Enum.HttpConstants.*;
+import static Server.Enum.Pages.*;
+
+public class Server implements Runnable {
+    private static String hostname = "localhost"; // Change this if we actually host this on a domain. -Kyle
+    private static final int backlog = -1; // maximum number of connections allowed in queue. >=0 values set value to OS-specific defualts. -Kyle
+    private static final int serverCloseDelay = 2; // how many seconds to wait for connections to close themselves after server stop
+    private static int port = 8081; // Defualt port for http servers is 8080 -Kyle
+    private static final String localContext = "./"; // File context on local machine to serve requested files (NOT related to Java imports, this is for HTTP connections) -Kyle
+    private static final String serverContext = "/"; // the root context to for the server to respond to when listening for requests. (the part that needs to go at the end of the URL) -Kyle
+    
+    private static HttpServer server;
+    public Server(final String hostname, final int port) throws IOException {
+        this.hostname = hostname;
+        this.port = port;
+        server = HttpServer.create(new InetSocketAddress(hostname, port), backlog);
+        this.server.setExecutor(null); // creates a default executor
+    }
+    
+    private static boolean isSocketUsed(final String hostname, final int port) { 
+    try (Socket s = new Socket(hostname, port)) {
+        return true;
+    } catch (IOException ex) {
+        // ignore, means that socket is occupied already
+    }
+    return false;
+}
+    
+    public static boolean isRunning() {
+        return isSocketUsed(hostname, port);
+    }
+    
+    public static String getLocalContext() {
+        return localContext;
+    }
+    
+    public static String getServerContext() {
+        return serverContext;
+    }
+    
+    public static int getPort() {
+        return port;
+    }
+    
+    public static String getHostname() {
+        return hostname;
+    }
+    
+    public static void stop() {
+        try {
+            server.stop(serverCloseDelay);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    @Override
+    public void run() {
+        try {
+            this.server.createContext(this.serverContext, new ServerConnectionHandler());
+            this.server.start();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    public static void main(final String... args) throws IOException {

```

Comment:
Main method should _never*_ be in the middle of the code

Reply:
👍